include/soci/soci.h

```diff
@@ -58,6 +59,8 @@
 // C++17
 #ifdef SOCI_HAVE_CXX17
 #include "soci/std-optional.h"
+#include "soci/std-ptr.h"
+#include "soci/std-path.h"
```

Comment:
Yes, I understand. In that case, maybe I should only include the headers for smart pointers? Because even though both smart pointers and std::filesystem::path are "convenient objects", std::filesystem::path seems to me more optional and project-specific than smart pointers.